.github/workflows/build-m1-binaries.yml

```diff
@@ -0,0 +1,121 @@
+name: Build on M1
+on:
+  pull_request:
+    paths:
+      - .github/workflows/build-m1-binaries.yml
+  push:
+    branches:
+      - nightly
+      - main
+  workflow_dispatch:
+jobs:
+  build_wheels:
+    name: "Build TorchVision M1 wheels"
+    runs-on: macos-m1
+    strategy:
+      matrix:
+        py_vers: [ "3.8", "3.9", "3.10" ]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build TorchVision M1 wheel
+        shell: arch -arch arm64 bash {0}
+        env:
+          ENV_NAME: conda-env-${{ github.run_id }}
+          PY_VERS: ${{ matrix.py_vers }}
+        run: |
+          . ~/miniconda3/etc/profile.d/conda.sh
+          # Needed for JPEG library detection as setup.py detects conda presence by running `shlex.which('conda')`
+          export PATH=~/miniconda3/bin:$PATH
+          set -ex
+          . packaging/pkg_helpers.bash
+          setup_build_version
+          WHL_NAME=torchvision-${BUILD_VERSION}-cp${PY_VERS/.}-cp${PY_VERS/.}-macosx_11_0_arm64.whl
+          conda create -yp ${ENV_NAME} python=${PY_VERS} numpy libpng jpeg wheel pkg-config
+          conda run -p ${ENV_NAME} python3 -mpip install torch --pre -f https://download.pytorch.org/whl/test/cpu/torch_test.html
```

Comment:
I have edited this line to rely on the test repo instead of the nightlies, but there are still various references to "nightly" in this script and I'm not sure what to do with them.

@malfet would you mind double checking this whole file?
(perhaps you'd like to wait after https://github.com/pytorch/vision/pull/6140?)